pennylane/ops/op_math/decompositions/controlled_decompositions.py

```diff
@@ -497,16 +504,32 @@ def _mcx_many_borrowed_workers(wires, **kwargs):
 
 
 def _mcx_two_workers_condition(num_control_wires, num_work_wires, **__):
-    return num_control_wires > 2 and num_work_wires >= 2
+    return num_control_wires > 2 and (
+        num_work_wires >= 2 or (num_work_wires == 1 and num_control_wires < 6)
+    )
 
 
 def _mcx_two_workers_resource(num_control_wires, work_wire_type, **__):
+
+    is_small_mcx = num_control_wires < 6
+
     if work_wire_type == "zeroed":
         n_ccx = 2 * num_control_wires - 3
-        return {ops.Toffoli: n_ccx, ops.X: n_ccx - 3 if num_control_wires < 6 else n_ccx - 5}
+        n_temporary_ccx_pairs = 2 - is_small_mcx
+        return {
+            ops.Toffoli: n_ccx - 2 * n_temporary_ccx_pairs,
+            ops.X: n_ccx - 3 if is_small_mcx else n_ccx - 5,
+            qml.TemporaryAND: n_temporary_ccx_pairs,
+            adjoint_resource_rep(qml.TemporaryAND): n_temporary_ccx_pairs,
```

Comment:
For each `Toffoli` pair that computes a temporary AND result, replace two `Toffoli` gates by one `TemporaryAND` and one `adjoint(TemporaryAND)`.